NumLib/DOF/LocalToGlobalIndexMap.cpp

```diff
@@ -194,51 +194,66 @@ LocalToGlobalIndexMap::LocalToGlobalIndexMap(
+          to_cumulative(std::vector<unsigned>(1, 1)))  // Single variable only.
 {
-    // There is only one mesh_subsets in the vector _mesh_subsets.
-    assert(_mesh_subsets.size() == 1);
-    auto const mss = *_mesh_subsets.front();
-
-    // For all MeshSubset in mesh_subsets and each element of that MeshSubset
-    // save a line of global indices.
-    for (MeshLib::MeshSubset const* const ms : mss)
+    // Each subset in the mesh_subsets represents a single component.
+    if (_mesh_subsets.size() != global_component_ids.size())
+        OGS_FATAL(
+            "Number of mesh subsets is not equal to number of components. "
+            "There are %d mesh subsets and %d components.",
+            mesh_subsets.size(), global_component_ids.size());
+
+    for (std::size_t i = 0; i < global_component_ids.size(); ++i)
     {
-        std::size_t const mesh_id = ms->getMeshID();
+        auto const& mss = *_mesh_subsets[i];
+
+        // For all MeshSubset in mesh_subsets and each element of that
+        // MeshSubset
+        // save a line of global indices.
+        for (MeshLib::MeshSubset const* const ms : mss)
+        {
+            std::size_t const mesh_id = ms->getMeshID();
 
-        findGlobalIndices(elements.cbegin(), elements.cend(), ms->getNodes(), mesh_id,
-                          component_id, 0);  // There is only one component to
-                                             // write out, therefore the zero
-                                             // parameter.
+            findGlobalIndices(elements.cbegin(), elements.cend(),
+                              ms->getNodes(), mesh_id, global_component_ids[i],
+                              i);
+        }
     }
 }
 
 LocalToGlobalIndexMap* LocalToGlobalIndexMap::deriveBoundaryConstrainedMap(
     int const variable_id,
-    int const component_id,
+    std::vector<int> const& component_ids,
     std::unique_ptr<MeshLib::MeshSubsets>&& mesh_subsets,
     std::vector<MeshLib::Element*> const& elements) const
 {
     DBUG("Construct reduced local to global index map.");
     // Create a subset of the current mesh component map.
-    auto const global_component_id =
-        getGlobalComponent(variable_id, component_id);
+    std::vector<std::size_t> global_component_ids;
+
+    for (auto component_id : component_ids)
+        global_component_ids.push_back(
+            getGlobalComponent(variable_id, component_id));
 
     auto mesh_component_map =
-        _mesh_component_map.getSubset(global_component_id, *mesh_subsets);
+        _mesh_component_map.getSubset(global_component_ids, *mesh_subsets);
 
+    // Create copies of the mesh_subsets for each of the global components.
     std::vector<std::unique_ptr<MeshLib::MeshSubsets>> all_mesh_subsets;
-    all_mesh_subsets.emplace_back(std::move(mesh_subsets));
+    for (std::size_t i = 0; i < global_component_ids.size(); ++i)
+        all_mesh_subsets.emplace_back(
+            new MeshLib::MeshSubsets{*mesh_subsets});
```

Comment:
- Why a copy? In the original code it was a `std::move`?
- Could the for loop also be written as ```for (auto global_component_id : global_component_ids)```?